pyam/core.py

```diff
@@ -489,34 +489,69 @@ def pivot_table(self, index, columns, values='value',
         df = df.unstack(level=columns, fill_value=fill_value)
         return df
 
-    def interpolate(self, time):
-        """Interpolate missing values in timeseries (linear interpolation)
+    def interpolate(self, time, **kwargs):
+        """Interpolate missing values in timeseries (linear interpolation by
+        default). Internally, :meth:`pandas.DataFrame.interpolate` is used.
 
         Parameters
         ----------
-        time : int or datetime
+        time : int or datetime, or list-like thereof
              Year or :class:`datetime.datetime` to be interpolated.
              This must match the datetime/year format of `self`.
+        inplace : bool, default False
+            if True, do operation inplace and return None
+        kwargs
+            passed to :meth:`pandas.DataFrame.interpolate`
         """
-        if self.time_col == 'year' and not isinstance(time, int):
+        ##
+        # TODO remove, and add kwarg inplace=False in next release
+        if 'inplace' not in kwargs:
+            deprecation_warning(
+                'Behavior of `interpolate` will change default to `False` in '
+                'a future release. Explicitly set it to avoid this warning.'
+            )
+            inplace = True
+        else:
+            inplace = kwargs.pop('inplace')
+        ##
+
+        # setup
+        ret = self.copy() if not inplace else self
+        interp_kwargs = dict(method='slinear', axis=1)
+        interp_kwargs.update(kwargs)
+        time = list(time) if islistable(time) else [time]
+        # TODO - have to explicitly cast to numpy datetime to sort later,
+        # could enforce as we do for year below
+        if self.time_col == 'time':
+            time = list(map(np.datetime64, time))
```

Comment:
@znicholls I had to convert to `np.datetime` here in order to make calls to `np.sort` and `np.unique` later on - numpy and datetime objects didn't play nice together. If you have a better suggestion let me know, but I observed that the datetime pyam object internally stores as `np.datetime`, so had to convert input.

Reply:
Yes the datetime vs. np.datetime difference is a constant source of annoyance so this casting is unavoidable I think. Could potentially be removed from pyam if the backend were xarray (it has better support for all sorts of time axis handling than pandas), but that's a job for another day.

Reply:
👍